cpp/src/binaryop/binaryop.cpp

```diff
@@ -419,25 +423,50 @@ bool is_same_scale_necessary(binary_operator op)
 std::unique_ptr<column> fixed_point_binary_operation(scalar const& lhs,
                                                      column_view const& rhs,
                                                      binary_operator op,
+                                                     thrust::optional<cudf::data_type> output_type,
                                                      rmm::cuda_stream_view stream,
                                                      rmm::mr::device_memory_resource* mr)
 {
   using namespace numeric;
 
   CUDF_EXPECTS(is_supported_fixed_point_binop(op), "Unsupported fixed_point binary operation");
-  CUDF_EXPECTS(lhs.type().id() == rhs.type().id(),
-               "Both columns must be of the same fixed_point type");
+  CUDF_EXPECTS(lhs.type().id() == rhs.type().id(), "Both columns must be of the same type");
```

Comment:
Use "type mismatch" for consistency with existing messages. Also they aren't both columns.

```suggestion
  CUDF_EXPECTS(lhs.type().id() == rhs.type().id(), "Data type mismatch");
```